src/env.d.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+type User = import('./types.ts').User
+
+type Action = (event: MouseEvent, user: User) => void
+
+type Enabled = (user: User) => boolean
+
+type RegisterAction = (icon: string, text: string, action: Action, enabled: Enabled) => void
+
+declare global {
+	interface Window {
+		OCA: {
+			Settings: {
+				UserList: {
+					registerAction: RegisterAction,
```

Comment:
Unrelated but:
IMHO this needs to be ocp or in a package. 
Having this only unofficially available feels wrong

Reply:
Agreed, can be split off https://github.com/nextcloud/server/issues/45937